must-gather/collection-scripts/gather_ceph_resources

```diff
@@ -103,31 +101,31 @@ for ns in $(oc get cephcluster --all-namespaces --no-headers | awk '{print $1}')
         for ((i = 0; i < ${#ceph_commands[@]}; i++)); do
             printf "collecting command output for: %s\n"  "${ceph_commands[$i]}"
             COMMAND_OUTPUT_FILE=${COMMAND_OUTPUT_DIR}/${ceph_commands[$i]// /_}
-            oc -n ${ns} exec ${HOSTNAME}-helper -- ${ceph_commands[$i]} --connect-timeout=15 >> ${COMMAND_OUTPUT_FILE}
+            timeout 30 oc -n "${ns}" exec "${HOSTNAME}"-helper -- ${ceph_commands[$i]} --connect-timeout=15 >> "${COMMAND_OUTPUT_FILE}"
             JSON_COMMAND_OUTPUT_FILE=${COMMAND_JSON_OUTPUT_DIR}/${ceph_commands[$i]// /_}_--format_json-pretty
-            oc -n ${ns} exec ${HOSTNAME}-helper -- ${ceph_commands[$i]} --connect-timeout=15 --format json-pretty >> ${JSON_COMMAND_OUTPUT_FILE}
+            timeout 30 oc -n "${ns}" exec "${HOSTNAME}"-helper -- ${ceph_commands[$i]} --connect-timeout=15 --format json-pretty >> "${JSON_COMMAND_OUTPUT_FILE}"
         done
     fi
 
     # Collecting output of ceph volume commands
     for ((i = 0; i < ${#ceph_volume_commands[@]}; i++)); do
         printf "collecting command output for: %s\n"  "${ceph_volume_commands[$i]}"
-        for osdPod in $(oc get pods -n ${ns} -l app=rook-ceph-osd --no-headers | awk '{print $1}'); do
-            pod_status=`oc get po ${osdPod} -n ${ns} -o jsonpath='{.status.phase}'`
+        for osdPod in $(oc get pods -n "${ns}" -l app=rook-ceph-osd --no-headers | awk '{print $1}'); do
+            pod_status=$(oc get po "${osdPod}" -n "${ns}" -o jsonpath='{.status.phase}')
             if [ "${pod_status}" != "Running" ]; then
                 continue
             fi
             COMMAND_OUTPUT_FILE=${COMMAND_OUTPUT_DIR}/${ceph_volume_commands[$i]// /_}
-            oc -n ${ns} exec ${osdPod} -- ${ceph_volume_commands[$i]} >> ${COMMAND_OUTPUT_FILE}
+            timeout 30 oc -n "${ns}" exec "${osdPod}" -- ${ceph_volume_commands[$i]} >> "${COMMAND_OUTPUT_FILE}"
         done
     done
     
     # Collecting ceph prepare volume logs
     for node in $(oc get nodes -l cluster.ocs.openshift.io/openshift-storage='' --no-headers | grep -w 'Ready' | awk '{print $1}'); do
         printf "collecting prepare volume logs from node %s \n"  "${node}"
         NODE_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/namespaces/${ns}/osd_prepare_volume_logs/${node}
-        mkdir -p ${NODE_OUTPUT_DIR}
-        oc debug nodes/${node} -- bash -c "test -f /host/var/lib/rook/log/${ns}/ceph-volume.log && cat /host/var/lib/rook/log/${ns}/ceph-volume.log" > ${NODE_OUTPUT_DIR}/ceph-volume.log
+        mkdir -p "${NODE_OUTPUT_DIR}"
+        oc debug nodes/"${node}" -- bash -c "test -f /host/var/lib/rook/log/${ns}/ceph-volume.log && cat /host/var/lib/rook/log/${ns}/ceph-volume.log" > "${NODE_OUTPUT_DIR}"/ceph-volume.log
```

Comment:
i think we should have a timeout here aswell.

Reply:
ok I will add it here also